subsys/bluetooth/mesh/sensor.c

```diff
@@ -623,7 +623,7 @@ const char *bt_mesh_sensor_ch_str(const struct sensor_value *ch)
 {
 	static char str[BT_MESH_SENSOR_CH_STR_LEN];
 
-	bt_mesh_sensor_ch_to_str(ch, str, sizeof(str));
+	__ASSERT_NO_MSG(bt_mesh_sensor_ch_to_str(ch, str, sizeof(str)) == 0);
```

Comment:
I'm not sure that API that parses customer data should fall into assertion. The overall devices become fragile due to customer incoming data.

Should it return a predefined string instead? Smth like:
```
const char *bt_mesh_sensor_ch_str(const struct sensor_value *ch)
{
	static char str[BT_MESH_SENSOR_CH_STR_LEN];

	if (bt_mesh_sensor_ch_to_str(ch, str, sizeof(str)) == 0) {
		strcpy(str, "Unknown format");
	}

	return str;
}
```